### PR TITLE
fix: [CDS-83555]: Update Shell Script optional configuration

### PR DESCRIPTION
**Pull request by pankajsingh-harness**
### Summary

**Issue:** Pipeline asks for reconcile when the user made the onDelegate field as runtime input in Shell Script Template and chooses executionTarget while using the template in Pipeline. This was happening as onDelegate field is removed from yaml and executionTarget field was added. 

**Fix:** Remove the onDelegate field completely from YAML and use executionTarget field only for different scenarios.

- executionTarget: {} // show OnDelegate
- executionTarget: <+inout> // Shop Execution Target as runtime fied
- executionTarget: { // Should Child field based on value
        connectorRef: <+input>
        host: <+input>
        workingDirectory: <+input>
  }
  
 YAML Before:
 ```
 - step:
    type: ShellScript
    name: ShellScript_1
    identifier: ShellScript_1
    spec:
      shell: Bash
      onDelegate: <+input>
      source:
        type: Inline
        spec:
          script: Echo Stage 4
      environmentVariables: []
      outputVariables: []
    timeout: 10m
```

YAML After:
```
- step:
    type: ShellScript
    name: ShellScript_1
    identifier: ShellScript_1
    spec:
      shell: Bash
      executionTarget: <+input>
      source:
        type: Inline
        spec:
          script: echo Hello World
      environmentVariables: []
      outputVariables: []
    timeout: 10m
```

There is no visual change.

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

File: src/modules/75-cd/components/BaseScript/BaseScript.tsx [hunk 1]
```diff
@@ -153,9 +164,6 @@ export default function BaseScript(props: {
           style={{ marginBottom: 0, marginTop: 0 }}
           disableTypeSelection={false}
           supportListOfExpressions={true}
-          onTypeChange={() => {
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/BaseScript/BaseScript.tsx:156]: This is not needed as `spec.source.spec.script` is updated once user choose  `Select script location`

File: src/modules/70-pipeline/components/PipelineSteps/Steps/CustomApproval/OptionalConfiguration.tsx [hunk 2]
```diff
@@ -23,17 +22,6 @@ import { scriptInputType, CustomApprovalFormData, CustomApprovalStepVariable } f
 import stepCss from '@pipeline/components/PipelineSteps/Steps/Steps.module.scss'
 import css from './CustomApproval.module.scss'
 
-export const targetTypeOptions: IOptionProps[] = [
```
Comment (pankajsingh-harness) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/CustomApproval/OptionalConfiguration.tsx:26]: Removed this un-used code.

File: src/modules/72-templates-library/components/TemplateStudio/__tests__/stateMock.ts [hunk 3]
```diff
@@ -74,9 +74,9 @@ export const secretManagerTemplateMock: NGTemplateInfoConfig = {
   spec: {
     shell: 'Bash',
     source: {
+      type: 'Inline',
```
Comment (pankajsingh-harness) [inline src/modules/72-templates-library/components/TemplateStudio/__tests__/stateMock.ts:77]: This initial value was wrong as per interface and formik structure. So Just updated that

File: src/modules/75-cd/components/PipelineSteps/ShellScriptStep/ShellScriptWidget.tsx [hunk 4]
```diff
@@ -106,64 +107,49 @@ export function ShellScriptWidget(
...
+            })
           })
-        })
+        }
       })
     }),
     ...getNameAndIdentifierSchema(getString, stepViewType)
   })
 
-  const values: ShellScriptFormData = {
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/PipelineSteps/ShellScriptStep/ShellScriptWidget.tsx:149]: This data processing is not needed as we are already doing this in the step itself

**Comment by pankajsingh-harness** [comment]
trigger messagecheck

**Review by sonaligoyal-harness — DISMISSED** [review]

File: src/modules/75-cd/components/BaseScript/BaseScript.tsx [hunk 5]
```diff
@@ -60,11 +60,16 @@ export default function BaseScript(props: {
...
+            const onDelegatePath = 'spec.onDelegate'
+            // Using setValues to ensure both values are updated simultaneously
             setValues(
               produce(formValues, draft => {
-                set(draft, 'spec.onDelegate', true)
+                // Update executionTarget target instead of onDelegate as we are removing onDelegate field from YAML
+                set(draft, 'spec.executionTarget', {})
                 set(draft, 'spec.shell', value.value)
+                if (!isUndefined(get(draft, onDelegatePath))) {
+                  set(draft, onDelegatePath, true)
```
[the remark this answers is not in the document]
Comment (pankajsingh-harness) [reply src/modules/75-cd/components/BaseScript/BaseScript.tsx:71]: @chetanvekariya-harness This is for backward compatibility to not trigger reconcile while editing already created data with the old field

**Review by chetanvekariya-harness — APPROVED** [review]

**Review by sonaligoyal-harness — APPROVED** [review]

**Review by sunnykesh-harness — APPROVED** [review]

**Pull request merged**